Correlation

In Lesson 2, we looked at a scatterplot of:

    area -> price

and we could visually say:

"Larger properties generally seem to have higher prices."

But that's a visual judgment.

Now we want to turn:

"It looks like they move together."

into:

"They have a Pearson correlation of r=0.72."

That number gives us a standardized way to describe the direction and strength of a linear relationship.


You started with:

One scatterplot has a tight cluster around a rising line. Another is a diffuse cloud that slopes upward. Which is stronger?

Exactly what Pearson answers.

Imagine:

Plot A
price
  │
  │             •
  │          •
  │       •
  │    •
  │ •
  └──────────────── area

Very tight around an upward line.

You might get:

r=0.95


Plot B
price
  │       •       •
  │  •
  │             •
  │    •    •
  │       •
  └──────────────── area

Still generally upward, but much more scattered.

Maybe:

r=0.45

Both are positive.

But:

|0.95|>|0.45|

Therefore the first relationship is stronger in the linear sense Pearson measures.

The sign tells the direction while the absolute value tells the strength


In [5]:
import pandas as pd

data_path = "../data/mexico-real-estate-combined-clean.csv"

data = pd.read_csv(data_path)


corr_price_area = data["price_usd"].corr(data["area_m2"])
full_corr = data.corr(numeric_only=True)
print(corr_price_area)
print(full_corr)


0.5855182454266903
                lat       lon   area_m2  price_usd
lat        1.000000 -0.480231  0.082314  -0.085868
lon       -0.480231  1.000000  0.095522   0.032815
area_m2    0.082314  0.095522  1.000000   0.585518
price_usd -0.085868  0.032815  0.585518   1.000000


What it means precisely:

The Pearson r of approximately 0.59 between area_m2 and price_usd tells us that roughly 35% of the variance in price can be "explained" by area (because r² = 0.59² ≈ 0.35). In plain language: if you know nothing about a property except its size, knowing the area reduces your uncertainty about the price by about 35%. The remaining 65% of price variation is explained by other factors — location, property type, condition, amenities, and factors we have not measured.

What it implies for prediction accuracy:

If you were to build the simplest possible predictive model — a straight line through the cleaned scatterplot — that model would have substantial predictive power but large individual errors. Concretely:

The model would correctly rank properties by price in about 70% of pairwise comparisons (this is related to a statistic called the concordance index).
But for any individual property, the prediction interval would be very wide — perhaps ±40,000 around the predicted value. Knowing a property is 120 m² tells you the predicted price is around 102,000, but the true price could easily be anywhere from 60,000 to 180,000.
This wide prediction interval is not a failure of the model — it is an accurate statement about how much unexplained variation exists. Area is a useful but incomplete predictor of price.
What r = 0.59 does not answer:

Is the relationship the same everywhere? A national r of 0.59 is a weighted average of all state-level relationships. Some states may have r ≈ 0.85 (size strongly predicts price), while others have r ≈ 0.10 (size barely matters). The national number conceals this variation entirely. Section 4 will reveal it.

Is the relationship causal? Correlation is not causation. The fact that larger properties cost more does not mean that adding square footage causes a property to become more valuable. Both area and price may be driven by the same underlying variable — location. In an expensive neighborhood, both larger properties and higher prices are the norm; in a cheap neighborhood, both smaller properties and lower prices are the norm. The correlation may partly reflect location sorting rather than a pure size effect.

Is the relationship linear? Pearson r assumes linearity. If the true relationship is curved — for example, if each additional square meter becomes less valuable as total size grows (diminishing returns) — then r will understate the true relationship. Section 5 will test this with the price_per_m2 feature.